sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1276,6 +1276,24 @@ class Analyzer(override val catalogManager: CatalogManager)
           AppendData.byPosition(r, query)
         } else if (conf.partitionOverwriteMode == PartitionOverwriteMode.DYNAMIC) {
           OverwritePartitionsDynamic.byPosition(r, query)
+        } else if (i.replacePredicates.nonEmpty) {
+          def findAttrInRelation(name: String): Attribute = {
```

Comment:
we should check with OSS reviewers(once this PR is ready) if there is some existing method for this, so that we don't need to duplicate code.

Reply:
add comments here since this is not obvious.